File: app/src/main/java/com/mdmobile/pocketconsole/dataModels/api/devices/IosDevice.java
package com.mdmobile.pocketconsole.dataModels.api.devices;

/**
 * Represent gson class for iOS device
 */

public class IosDevice extends BasicDevice {
    private String AgentVersion, BluetoothMACAddress, BuildVersion, CarrierSettingsVersion, CellularCarrier,
            CurrentMCC, CurrentMNC, FirmwareVersion, CellularTechnology, ExchangeStatus, ICCID, IMEI_MEID_ESN,
            Ipv6, LastCheckInTime, LastAgentConnectTime, LastAgentDisconnectTime, LastLoggedOnUser, LastStatusUpdate,
            ManufacturerSerialNumber, ModelNumber, ModemFirmwareVersion, PersonalizedName, PhoneNumber, ProductName,
            SIMCarrierNetwork, SubscriberMCC, SubscriberMNC, SubscriberNumber, UserIdHash, PasscodeStatus;

    private int HardwareEncryptionCaps, NetworkConnectionType;
    private short BatteryStatus;

    private boolean InRoaming, DataRoamingEnabled, IsAgentCompatible, IsAgentless, IsDeviceLocatorServiceEnabled,
            IsDoNotDisturbInEffect, IsEncrypted, IsEnrolled, IsITunesStoreAccountActive, IsOSSecure,
            IsPersonalHotspotEnabled, IsSupervised, PasscodeEnabled, VoiceRoamingEnabled, ExchangeBlocked;


//    TODO:Support following info
//    DeviceTerms	DeviceTerms
//    DeviceUser	DeviceUserInfo


    public IosDevice(String Kind, String deviceId, String deviceName, String enrollmentTime, String family, String
            hostName, String MACAddress, String manufacturer, String mode, String model,
                     String OSVersion, String path, Boolean complianceStatus,
                     Boolean isAgentOnline, Boolean isVirtual, String agentVersion, String bluetoothMACAddress,
                     String buildVersion, String carrierSettingsVersion, String cellularCarrier, String currentMCC,
                     String currentMNC, String firmwareVersion, String cellularTechnology, String exchangeStatus, String ICCID,
                     String IMEI_MEID_ESN, String ipv6, String lastCheckInTime, String lastAgentConnectTime,
                     String lastAgentDisconnectTime, String lastLoggedOnUser, String lastStatusUpdate, String manufacturerSerialNumber,
                     String modelNumber, String modemFirmwareVersion, String personalizedName, String phoneNumber,
                     String productName, String SIMCarrierNetwork, String subscriberMCC, String subscriberMNC,
                     String subscriberNumber, String userIdHash, String passcodeStatus, int hardwareEncryptionCaps,
                     int networkConnectionType, short batteryStatus, boolean inRoaming, boolean dataRoamingEnabled,
                     boolean isAgentCompatible, boolean isAgentless, boolean isDeviceLocatorServiceEnabled,
                     boolean isDoNotDisturbInEffect, boolean isEncrypted, boolean isEnrolled, boolean isITunesStoreAccountActive,
                     boolean isOSSecure, boolean isPersonalHotspotEnabled, boolean isSupervised, boolean passcodeEnabled,
                     boolean voiceRoamingEnabled, boolean exchangeBlocked, String Platform) {

        super(Kind, deviceId, deviceName, enrollmentTime, family, hostName, MACAddress, manufacturer, mode, model,
                OSVersion,
                path, complianceStatus, isAgentOnline, isVirtual, Platform);

        AgentVersion = agentVersion;
        BluetoothMACAddress = bluetoothMACAddress;
        BuildVersion = buildVersion;
        CarrierSettingsVersion = carrierSettingsVersion;
        CellularCarrier = cellularCarrier;
        CurrentMCC = currentMCC;
        CurrentMNC = currentMNC;
        FirmwareVersion = firmwareVersion;
        CellularTechnology = cellularTechnology;
        ExchangeStatus = exchangeStatus;
        this.ICCID = ICCID;
        this.IMEI_MEID_ESN = IMEI_MEID_ESN;
        Ipv6 = ipv6;
        LastCheckInTime = lastCheckInTime;
        LastAgentConnectTime = lastAgentConnectTime;
        LastAgentDisconnectTime = lastAgentDisconnectTime;
        LastLoggedOnUser = lastLoggedOnUser;
        LastStatusUpdate = lastStatusUpdate;
        ManufacturerSerialNumber = manufacturerSerialNumber;
        ModelNumber = modelNumber;
        ModemFirmwareVersion = modemFirmwareVersion;
        PersonalizedName = personalizedName;
        PhoneNumber = phoneNumber;
        ProductName = productName;
        this.SIMCarrierNetwork = SIMCarrierNetwork;
        SubscriberMCC = subscriberMCC;
        SubscriberMNC = subscriberMNC;
        SubscriberNumber = subscriberNumber;
        UserIdHash = userIdHash;
        PasscodeStatus = passcodeStatus;
        HardwareEncryptionCaps = hardwareEncryptionCaps;
        NetworkConnectionType = networkConnectionType;
        BatteryStatus = batteryStatus;
        InRoaming = inRoaming;
        DataRoamingEnabled = dataRoamingEnabled;
        IsAgentCompatible = isAgentCompatible;
        IsAgentless = isAgentless;
        IsDeviceLocatorServiceEnabled = isDeviceLocatorServiceEnabled;
        IsDoNotDisturbInEffect = isDoNotDisturbInEffect;
        IsEncrypted = isEncrypted;
        IsEnrolled = isEnrolled;
        IsITunesStoreAccountActive = isITunesStoreAccountActive;
        IsOSSecure = isOSSecure;
        IsPersonalHotspotEnabled = isPersonalHotspotEnabled;
        IsSupervised = isSupervised;
        PasscodeEnabled = passcodeEnabled;
        VoiceRoamingEnabled = voiceRoamingEnabled;
        ExchangeBlocked = exchangeBlocked;
    }

    public String getAgentVersion() {
        return AgentVersion;
    }

    public String getBluetoothMACAddress() {
        return BluetoothMACAddress;
    }

    public String getBuildVersion() {
        return BuildVersion;
    }

    public String getCarrierSettingsVersion() {
        return CarrierSettingsVersion;
    }

    public String getCellularCarrier() {
        return CellularCarrier;
    }

    public String getCurrentMCC() {
        return CurrentMCC;
    }

    public String getCurrentMNC() {
        return CurrentMNC;
    }

    public String getFirmwareVersion() {
        return FirmwareVersion;
    }

    public String getCellularTechnology() {
        return CellularTechnology;
    }

    public String getExchangeStatus() {
        return ExchangeStatus;
    }

    public String getICCID() {
        return ICCID;
    }

    public String getIMEI_MEID_ESN() {
        return IMEI_MEID_ESN;
    }

    public String getIpv6() {
        return Ipv6;
    }

    public String getLastCheckInTime() {
        return LastCheckInTime;
    }

    public String getLastAgentConnectTime() {
        return LastAgentConnectTime;
    }

    public String getLastAgentDisconnectTime() {
        return LastAgentDisconnectTime;
    }

    public String getLastLoggedOnUser() {
        return LastLoggedOnUser;
    }

    public String getLastStatusUpdate() {
        return LastStatusUpdate;
    }

    public String getManufacturerSerialNumber() {
        return ManufacturerSerialNumber;
    }

    public String getModelNumber() {
        return ModelNumber;
    }

    public String getModemFirmwareVersion() {
        return ModemFirmwareVersion;
    }

    public String getPersonalizedName() {
        return PersonalizedName;
    }

    public String getPhoneNumber() {
        return PhoneNumber;
    }

    public String getProductName() {
        return ProductName;
    }

    public String getSIMCarrierNetwork() {
        return SIMCarrierNetwork;
    }

    public String getSubscriberMCC() {
        return SubscriberMCC;
    }

    public String getSubscriberMNC() {
        return SubscriberMNC;
    }

    public String getSubscriberNumber() {
        return SubscriberNumber;
    }

    public String getUserIdHash() {
        return UserIdHash;
    }

    public String getPasscodeStatus() {
        return PasscodeStatus;
    }

    public int getHardwareEncryptionCaps() {
        return HardwareEncryptionCaps;
    }

    public int getNetworkConnectionType() {
        return NetworkConnectionType;
    }

    public short getBatteryStatus() {
        return BatteryStatus;
    }

    public int isInRoaming() {
        return InRoaming ? 1 : 0;
    }

    public int isDataRoamingEnabled() {
        return DataRoamingEnabled ? 1 : 0;
    }

    public int isAgentCompatible() {
        return IsAgentCompatible ? 1 : 0;
    }

    public int isAgentless() {
        return IsAgentless ? 1 : 0;
    }

    public int isDeviceLocatorServiceEnabled() {
        return IsDeviceLocatorServiceEnabled ? 1 : 0;
    }

    public int isDoNotDisturbInEffect() {
        return IsDoNotDisturbInEffect ? 1 : 0;
    }

    public int isEncrypted() {
        return IsEncrypted ? 1 : 0;
    }

    public int isEnrolled() {
        return IsEnrolled ? 1 : 0;
    }

    public int isITunesStoreAccountActive() {
        return IsITunesStoreAccountActive ? 1 : 0;
    }

    public int isOSSecure() {
        return IsOSSecure ? 1 : 0;
    }

    public int isPersonalHotspotEnabled() {
        return IsPersonalHotspotEnabled ? 1 : 0;
    }

    public int isSupervised() {
        return IsSupervised ? 1 : 0;
    }

    public int isPasscodeEnabled() {
        return PasscodeEnabled ? 1 : 0;
    }

    public int isVoiceRoamingEnabled() {
        return VoiceRoamingEnabled ? 1 : 0;
    }

    public int isExchangeBlocked() {
        return ExchangeBlocked ? 1 : 0;
    }
}

File: app/src/main/java/com/mdmobile/pocketconsole/dataModels/api/devices/IosDevice.kt
package com.mdmobile.pocketconsole.dataModels.api.devices

import android.annotation.SuppressLint

/**
 * Represent gson class for iOS device
 */

@SuppressLint("ParcelCreator")
class IosDevice(Kind: String = "N/A", DeviceId: String = "N/A", DeviceName: String = "N/A", EnrollmentTime: String = "N/A",
                Family: String = "N/A", HostName: String = "N/A", MACAddress: String = "N/A", Manufacturer: String = "N/A",
                Mode: String = "N/A", Model: String = "N/A", OSVersion: String = "N/A", Path: String = "N/A",
                ComplianceStatus: Boolean = false, IsAgentOnline: Boolean = false,
                IsVirtual: Boolean = false, Platform: String = "N/A", val AgentVersion: String = "N/A", val BluetoothMACAddress: String = "N/A",
                val BuildVersion: String = "N/A", val CarrierSettingsVersion: String = "N/A", val CellularCarrier: String = "N/A", val CurrentMCC: String = "N/A",
                val CurrentMNC: String = "N/A", val FirmwareVersion: String = "N/A", val CellularTechnology: String = "N/A", val ExchangeStatus: String = "N/A", val ICCID: String = "N/A",
                val IMEI_MEID_ESN: String = "N/A", val Ipv6: String = "N/A", val LastCheckInTime: String = "N/A", val LastAgentConnectTime: String = "N/A",
                val LastAgentDisconnectTime: String = "N/A", val LastLoggedOnUser: String = "N/A", val LastStatusUpdate: String = "N/A", val ManufacturerSerialNumber: String = "N/A",
                val ModelNumber: String = "N/A", val ModemFirmwareVersion: String = "N/A", val PersonalizedName: String = "N/A", val PhoneNumber: String = "N/A",
                val ProductName: String = "N/A", val SimCarrierNetwork: String = "N/A", val SubscriberMCC: String = "N/A", val SubscriberMNC: String = "N/A",
                val SubscriberNumber: String = "N/A", val UserIdHash: String = "N/A", val PasscodeStatus: String = "N/A", val HardwareEncryptionCaps: Int = -1,
                val NetworkConnectionType: Int = -1, val BatteryStatus: Short = -1, private val InRoaming: Boolean = false, private val DataRoamingEnabled: Boolean = false,
                val IsAgentCompatible: Boolean = false, val IsAgentless: Boolean = false,
                val IsDeviceLocatorServiceEnabled: Boolean = false, val IsDoNotDisturbInEffect: Boolean = false,
                val IsEncrypted: Boolean = false, val IsEnrolled: Boolean = false, val IsITunesStoreAccountActive: Boolean = false,
                val IsOSSecure: Boolean = false, val IsPersonalHotspotEnabled: Boolean = false, val IsSupervised: Boolean = false,
                val PasscodeEnabled: Boolean = false, val VoiceRoamingEnabled: Boolean = false, val ExchangeBlocked: Boolean = false)
    : BasicDevice(Kind, DeviceId, DeviceName, EnrollmentTime, Family, HostName, MACAddress, Manufacturer, Mode, Model,
        OSVersion, Path, ComplianceStatus, IsAgentOnline, IsVirtual, Platform), IDevice<IosDevice> {

    //    TODO:Support following info
    //    DeviceTerms	DeviceTerms
    //    DeviceUser	DeviceUserInfo

    private val isInRoaming: Int
        get() = if (InRoaming) 1 else 0

    private val isDataRoamingEnabled: Int
        get() = if (DataRoamingEnabled) 1 else 0

    private val isAgentCompatible: Int
        get() = if (IsAgentCompatible) 1 else 0

    private val isAgentless: Int
        get() = if (IsAgentless) 1 else 0

    private val isDeviceLocatorServiceEnabled: Int
        get() = if (IsDeviceLocatorServiceEnabled) 1 else 0

    private val isDoNotDisturbInEffect: Int
        get() = if (IsDoNotDisturbInEffect) 1 else 0

    private val isEncrypted: Int
        get() = if (IsEncrypted) 1 else 0

    private val isEnrolled: Int
        get() = if (IsEnrolled) 1 else 0

    private val isITunesStoreAccountActive: Int
        get() = if (IsITunesStoreAccountActive) 1 else 0

    private val isOSSecure: Int
        get() = if (IsOSSecure) 1 else 0

    private val isPersonalHotspotEnabled: Int
        get() = if (IsPersonalHotspotEnabled) 1 else 0

    private val isSupervised: Int
        get() = if (IsSupervised) 1 else 0

    private val isPasscodeEnabled: Int
        get() = if (PasscodeEnabled) 1 else 0

    private val isVoiceRoamingEnabled: Int
        get() = if (VoiceRoamingEnabled) 1 else 0

    private val isExchangeBlocked: Int
        get() = if (ExchangeBlocked) 1 else 0

    override fun getDevice(): IosDevice {
        return this
    }

}
